### PR TITLE
add a --no-generate-docs flag that is faster and only displays warnings

### LINKED ISSUE #1537: Allow running without generating files – just to see errors
I have this working. I will land it as a series of PRs.

### PR DESCRIPTION
**Pull request by jcollins-g**
Fixes #1537.

Allows the user to use dartdoc to show dartdoc errors and warnings only without actually generating documentation.

Also has some changes to output in the common case.

Without --no-generate-docs:

```
~/dart/dartdoc
$ dart bin/dartdoc.dart
Documenting dartdoc...
parsing /usr/local/google/home/jcollins/dart/dartdoc/lib/dartdoc.dart...
parsing /usr/local/google/home/jcollins/dart/all_sdks/2.1.1-dev.1.0/lib/async/async.dart...
parsing /usr/local/google/home/jcollins/dart/all_sdks/2.1.1-dev.1.0/lib/collection/collection.dart...

[...]
parsing /usr/local/google/home/jcollins/.pub-cache/hosted/pub.dartlang.org/matcher-0.12.3+1/lib/mirror_matchers.dart...
Initialized dartdoc with 668 libraries in 47.0 seconds
documenting dartdoc
Generating docs for library dartdoc from package:dartdoc/dartdoc.dart...
  warning: unresolved doc reference [Property], from dartdoc.TopLevelContainer: (file:///usr/local/google/home/jcollins/dart/dartdoc/lib/src/model.dart:5464:16)
  warning: unresolved doc reference [_MarkdownCommentReference._getResultsForClass]
    from dartdoc.Accessor.namePart: (file:///usr/local/google/home/jcollins/dart/dartdoc/lib/src/model.dart)
    in documentation inherited from dartdoc.models.Nameable.namePart: (file:///usr/local/google/home/jcollins/dart/dartdoc/lib/src/model.dart:4548:14)
  warning: unresolved doc reference [_MarkdownCommentReference._getResultsForClass]
    from dartdoc.Class.allModelElementsByNamePart: (file:///usr/local/google/home/jcollins/dart/dartdoc/lib/src/model.dart)
    in documentation inherited from dartdoc.Class.allModelElementsByNamePart: (file:///usr/local/google/home/jcollins/dart/dartdoc/lib/src/model.dart:681:39)
  warning: unresolved doc reference [_MarkdownCommentReference._getResultsForClass]
    from dartdoc.Nameable.namePart: (file:///usr/local/google/home/jcollins/dart/dartdoc/lib/src/model.dart)
    in documentation inherited from dartdoc.Nameable.namePart: (file:///usr/local/google/home/jcollins/dart/dartdoc/lib/src/model.dart:4548:14)
  warning: unresolved doc reference [config.autoIncludeDependencies]
    from dartdoc.Package.isLocal: (file:///usr/local/google/home/jcollins/dart/dartdoc/lib/src/model.dart)
    in documentation inherited from dartdoc.Package.isLocal: (file:///usr/local/google/home/jcollins/dart/dartdoc/lib/src/model.dart:5889:12)
  warning: unresolved doc reference [toolPath]
    from dartdoc.Snapshot.needsSnapshot: (file:///usr/local/google/home/jcollins/dart/dartdoc/lib/src/dartdoc_options.dart)
    in documentation inherited from dartdoc.Snapshot.needsSnapshot: (file:///usr/local/google/home/jcollins/dart/dartdoc/lib/src/dartdoc_options.dart:209:12)
Validating docs...
found 6 warnings and 0 errors
Documented 1 public library in 7.6 seconds
Success! Docs generated into /usr/local/google/home/jcollins/dart/dartdoc/doc/api
```

With --no-generate-docs:
```
~/dart/dartdoc
$ dart bin/dartdoc.dart --no-generate-docs
Documenting dartdoc...
  warning: unresolved doc reference [_MarkdownCommentReference._getResultsForClass]
    from dartdoc.Accessor.namePart: (file:///usr/local/google/home/jcollins/dart/dartdoc/lib/src/model.dart)
    in documentation inherited from dartdoc.models.Nameable.namePart: (file:///usr/local/google/home/jcollins/dart/dartdoc/lib/src/model.dart:4548:14)
  warning: unresolved doc reference [_MarkdownCommentReference._getResultsForClass]
    from dartdoc.Nameable.namePart: (file:///usr/local/google/home/jcollins/dart/dartdoc/lib/src/model.dart)
    in documentation inherited from dartdoc.Nameable.namePart: (file:///usr/local/google/home/jcollins/dart/dartdoc/lib/src/model.dart:4548:14)
  warning: unresolved doc reference [_MarkdownCommentReference._getResultsForClass]
    from dartdoc.Class.allModelElementsByNamePart: (file:///usr/local/google/home/jcollins/dart/dartdoc/lib/src/model.dart)
    in documentation inherited from dartdoc.Class.allModelElementsByNamePart: (file:///usr/local/google/home/jcollins/dart/dartdoc/lib/src/model.dart:681:39)
  warning: unresolved doc reference [config.autoIncludeDependencies]
    from dartdoc.Package.isLocal: (file:///usr/local/google/home/jcollins/dart/dartdoc/lib/src/model.dart)
    in documentation inherited from dartdoc.Package.isLocal: (file:///usr/local/google/home/jcollins/dart/dartdoc/lib/src/model.dart:5889:12)
  warning: unresolved doc reference [toolPath]
    from dartdoc.Snapshot.needsSnapshot: (file:///usr/local/google/home/jcollins/dart/dartdoc/lib/src/dartdoc_options.dart)
    in documentation inherited from dartdoc.Snapshot.needsSnapshot: (file:///usr/local/google/home/jcollins/dart/dartdoc/lib/src/dartdoc_options.dart:209:12)
  warning: unresolved doc reference [Property], from dartdoc.TopLevelContainer: (file:///usr/local/google/home/jcollins/dart/dartdoc/lib/src/model.dart:5464:16)
found 6 warnings and 0 errors
```

### CODE REVIEW

File: lib/src/logging.dart [hunk 1]
```diff
@@ -96,13 +96,15 @@ void startLogging(LoggingContext config) {
...
-        } else {
-          print(message);
+        if (!config.quiet) {
+          if (config.showProgress && message.endsWith('...')) {
+            // Assume there may be more progress to print, so omit the trailing
+            // newline
+            writingProgress = true;
+            stdout.write(message);
+          } else {
+            print(message);
```
Comment (pq) [inline lib/src/logging.dart]: should this use `stdout` too?

**Comment by kevmoo** [comment]
@jcollins-g – any particular thing you want me to look at?

**Comment by jcollins-g** [comment]
@kevmoo I mostly put you on the list as you said you had an implementation of this in #1537 and so might be interested.

File: lib/src/logging.dart (hunk 1 above)
Comment (jcollins-g) [reply lib/src/logging.dart]: This code is making use of the side-effect of print that adds a `\n`.

**Pull request merged**